src/DefaultBuilder/src/WebApplication.cs

```diff
@@ -174,7 +174,12 @@ public void Run(string? url = null)
         RequestDelegate IApplicationBuilder.Build() => BuildRequestDelegate();
 
         // REVIEW: Should this be wrapping another type?
-        IApplicationBuilder IApplicationBuilder.New() => ApplicationBuilder.New();
+        IApplicationBuilder IApplicationBuilder.New()
+        {
+            var newBuilder = ApplicationBuilder.New();
+            newBuilder.Properties.Remove(GlobalEndpointRouteBuilderKey);
```

Comment:
nit: Add a comment about why we're doing this.

Reply:
We need to also update [here](https://github.com/dotnet/aspnetcore/pull/35336/files#diff-7baa2bd549d6a9332de6e302d1f40a0a7c6ad6404a8bb551d3a44bfa5abfe840R190).  Just change it to call this method so you can share the code that removes this key.